engines/catalog/config/routes.rb

```diff
@@ -1,2 +1,2 @@
-Catalog::Engine.routes.draw do
```

Comment:
I don't entirely agree with this change and the one in `config/routes.rb`. I don't like messing with the app's routes but prefer each engine to deal with its own in an isolated way from the rest. In the end, it's "just a gem" that could even be published to RubyGems. Then, the lines in `config/routes.rb`, the way I see is the explicit "installation" of those gems in our app. If we keep the analogy, just imagine using a gem that modifies your app. You would definitely complain.

Reply:
these are not gems, I dont think the analogy and idea of "publishing to rubygems" will ever work. these engines in the end will always depend on some core modules and they will always depend on each other to some extent (how would you do order management with products from catalog? I can see how the main app will disappear or become a simple "core module".
anyway, I agree with you on the routes and I think it would be better to have engine specific routes, but I couldnt make it work...
I dont see an alternative.

Reply:
so, would you agree if we say the routes like this are not ideal and we aim to switch back to engine specific routes (I'd create a tech debt issue for it) and we go forward like this for now?

Reply:
:+1: issue created: #5091 5122